src/ape/api/accounts.py

```diff
@@ -100,6 +103,36 @@ def sign_raw_msghash(self, msghash: "HexBytes") -> Optional[MessageSignature]:
             f"Raw message signing is not supported by '{self.__class__.__name__}'"
         )
 
+    def sign_authorization(
+        self,
+        address: Any,
+        chain_id: Optional[int] = None,
+    ) -> Optional[MessageSignature]:
+        """
+        Sign an `EIP-7702 <https://eips.ethereum.org/EIPS/eip-7702>`__ Authorization.
+
+        Args:
+          address (Any): A delegate address to sign the authorization for.
+          chain_id (Optional[int]):
+            The chain ID that the authorization should be valid for.
+            A value of ``0`` means that the authorization is valid for **any chain**.
+            Default tells implementation to use the currently connected network's ``chain_id``.
+
+        Returns:
+          :class:`~ape.types.signatures.MessageSignature` (optional):
+            The signature corresponding to the message.
+
+        ```{caution}
+        This action has the capability to be extremely destructive to the signer, and might lead to
+        full account compromise. All implementations are recommended to ensure that the signer be
+        made aware of the severity and impact of this action through some callout.
+        ```
+        """
+
+        raise NotImplementedError(
```

Comment:
use `APINotImplementedError` if it is common user facing expected error; it displays nicer.